ush/ocnpost.py

```diff
@@ -0,0 +1,646 @@
+
+        end subroutine
+    '''
+
+    try:
+        import esmfwghtinterp_f90subroutine
+    except ImportError or ModuleNotFoundError:
+        print('Weighted interpolation fortran subroutine has not been compiled. Compiling...')
+        np.f2py.compile(fsource1, modulename='esmfwghtinterp_f90subroutine', verbose=False)
+        import esmfwghtinterp_f90subroutine
+
+def esmfmanualregrid(src,S,row,col,frac_b,n_s,varmeth,slatd,slond,dlatd,dlond):
+    import esmfwghtinterp_f90subroutine 
+
+    vardims=len(src.shape)
+    if vardims == 4:
+        nlevs=40
+    if vardims == 3:
+        nlevs=1
+
+    dst=np.zeros(shape=(1,nlevs,dlatd,dlond),dtype=float)
+
+    for i in range(0,nlevs): 
+
+        if vardims == 4:
+            src_oned=np.reshape(np.transpose(src[0,i,:,:]),(slatd*slond),'F')
+        if vardims == 3:
+            src_oned=np.reshape(np.transpose(src[0,:,:]),(slatd*slond),'F')
+
+        if 1 == 0:
+            dst_oned= np.zeros(shape=(dlatd*dlond),dtype=float)
+            for m in range(0,n_s):
+                dst_oned[row[m]-1]=dst_oned[row[m]-1]+S[m]*src_oned[col[m]-1]
+
+        if 1 == 1:        
+            dst_oned=esmfwghtinterp_f90subroutine.bilin(src_oned,S,row,col,frac_b,dlatd*dlond)
+
+        dst[0,i,:,:] = np.transpose(np.reshape(dst_oned,(dlond,dlatd),'F'))
+    return dst
+
+def main():
+
+    print('Main program has started')
+
+    #Compile fortran subroutines if they have not already been compiled. 
+    #If already compiled, load fortran-related modules into Python
+    initfortranprog()
+
+    if len(sys.argv) == 6:
+        infiles0=sys.argv[1]
+        outfiles0=sys.argv[2]
+        nemsrc=sys.argv[3]
+        parmfile=sys.argv[4]
+        dstgrds=[sys.argv[5]]
+    else:
+        print('Incorrect number of arguments.')
+        exit()
+
+    if infiles0.endswith('.nc') and outfiles0.endswith('.nc'):
+        print('Input and output files are netcdf.')
+        infiles1=[sys.argv[1]]
+        outfiles1=[sys.argv[2]]
+        dimfiles=1
+    else:
+        with open(infiles0) as f1:
+            infiles1 = f1.read().splitlines()
+        with open(outfiles0) as f2:
+            outfiles1 = f2.read().splitlines()
+
+        dimfilesi=len(infiles1)
+        dimfileso=len(outfiles1)
+
+        if dimfilesi == dimfileso:
+            dimfiles=dimfilesi
+        else:
+            print('ERROR: Input filelist must be same length as output filelist.')
+            exit()
+
+    output_masks=False
+    methods=['bilinear','conserve']
+    dsttype=['rect.']
+    regridtype='manual' #manual or auto
+
+    #variables to be regridded with the native tripole stagger location
+    #and dimensionality 
+    #first BM contained only field "mld", which was actually ePBL
+    #the remaining BMs contain ePBL, MLD_003 and MLD_0125 
+    #the following NCO command will be issued at the end 
+    #to rename the variable mld to ePBL if the variable mld is found
+
+    ncocmd=['ncrename -O -v MLD_003,mld']
```

Comment:
Where is this used?  Remove unused codes

Reply:
Thanks for catching this. I will remove this since it will not be needed.